lib/js/vdom.js

```diff
@@ -0,0 +1,584 @@
+/*
+ * Virtual DOM integration.
+ */
+// Assumes virtual-dom.js has been loaded.
+
```

Comment:
For better encapulation you may want to wrap everything in a JavaScript-style module, that is
```
const VDom = (function() {
   function getElementID() { .. }
   return {'getElementID':getElementID, ...};
})();

// Expose the public members directly
const getElementID = VDom.getElementID;
...
```
At least this reduces the chance of a name collision until we have some proper design for encapsulating alien things.